shaders/commonProfileShaders.cpp

```diff
@@ -55,11 +56,17 @@ namespace GLTF
         ((std::istream*)png_get_io_ptr(pngPtr))->read((char*)data, length);
     }
 #endif
+	static std::unordered_map<std::string, bool> haveAlphas;		// Caching to remove up to 98% execution time
```

Comment:
a priori, I'm almost certain that's not true. Being ordered is the stronger property (requires comparison instead of just hashing, and we don't need to iterate over these in any deterministic order). Most ordered maps are going to have something like `O(m log n)` performance for most operations (where m is the length of the elements and n is the number of elements), and unordered maps are going to have `O(m)`. The only possible exception is if ordered map is implemented as a radix tree behind the scenes, it should also be `O(m)`, but with substantially worse cache locality properties due to pointer chasing (and I'm nearly certain the standard library isn't using a radix tree behind the scenes).

Random benchmarking results from Google back me up: http://supercomputingblog.com/windows/ordered-map-vs-unordered-map-a-performance-study/